dev-libs/spdlog/files/spdlog-unbundle-fmt.patch

```diff
@@ -0,0 +1,90 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index af1e398..e518a51 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -79,6 +79,7 @@ install(
+ install(
+     DIRECTORY "${HEADER_BASE}/${PROJECT_NAME}"
+     DESTINATION "${include_install_dir}"
++    PATTERN "bundled*" EXCLUDE
+ )
+ 
+ # install project version file
+diff --git a/include/spdlog/fmt/fmt.h b/include/spdlog/fmt/fmt.h
+index e2e0485..f07f8e5 100644
+--- a/include/spdlog/fmt/fmt.h
++++ b/include/spdlog/fmt/fmt.h
+@@ -10,24 +10,8 @@
+ // By default spdlog include its own copy.
+ //
+ 
+-#if !defined(SPDLOG_FMT_EXTERNAL)
+-
+-#ifndef FMT_HEADER_ONLY
+-#define FMT_HEADER_ONLY
+-#endif
+-#ifndef FMT_USE_WINDOWS_H
+-#define FMT_USE_WINDOWS_H 0
+-#endif
+-#include "bundled/format.h"
+-#if defined(SPDLOG_FMT_PRINTF)
+-#include "bundled/printf.h"
+-#endif
+-
+-#else // external fmtlib
+-
+ #include <fmt/format.h>
+ #if defined(SPDLOG_FMT_PRINTF)
+ #include <fmt/printf.h>
+ #endif
+ 
+-#endif
+diff --git a/include/spdlog/fmt/ostr.h b/include/spdlog/fmt/ostr.h
+index 9902898..c1a2223 100644
+--- a/include/spdlog/fmt/ostr.h
++++ b/include/spdlog/fmt/ostr.h
+@@ -7,12 +7,5 @@
+ //
+ // include bundled or external copy of fmtlib's ostream support
+ //
+-#if !defined(SPDLOG_FMT_EXTERNAL)
+-#ifndef FMT_HEADER_ONLY
+-#define FMT_HEADER_ONLY
+-#endif
+-#include "bundled/ostream.h"
+-#include "fmt.h"
+-#else
+ #include <fmt/ostream.h>
+-#endif
++
+diff --git a/include/spdlog/tweakme.h b/include/spdlog/tweakme.h
+index 50ad3fb..3b04a5d 100644
+--- a/include/spdlog/tweakme.h
++++ b/include/spdlog/tweakme.h
+@@ -88,13 +88,6 @@
+ // #define SPDLOG_EOL ";-)\n"
+ ///////////////////////////////////////////////////////////////////////////////
+ 
+-///////////////////////////////////////////////////////////////////////////////
+-// Uncomment to use your own copy of the fmt library instead of spdlog's copy.
+-// In this case spdlog will try to include <fmt/format.h> so set your -I flag accordingly.
+-//
+-// #define SPDLOG_FMT_EXTERNAL
+-///////////////////////////////////////////////////////////////////////////////
+-
+ ///////////////////////////////////////////////////////////////////////////////
+ // Uncomment to use printf-style messages in your logs instead of the usual
+ // format-style used by default.
+diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
+index d37f3c4..0b11c8b 100644
+--- a/tests/CMakeLists.txt
++++ b/tests/CMakeLists.txt
+@@ -18,7 +18,7 @@ set(SPDLOG_UTESTS_SOURCES
+ 
+ add_executable(${PROJECT_NAME} ${SPDLOG_UTESTS_SOURCES})
+ target_link_libraries(${PROJECT_NAME} PRIVATE Threads::Threads)
+-target_link_libraries(${PROJECT_NAME} PRIVATE spdlog)
++target_link_libraries(${PROJECT_NAME} PRIVATE spdlog fmt)
```

Comment:
That said, wasn't it supposed to be header-only?

Reply:
If fmt is bundled it's header-only, if not fmt source code is not compiled as part of spdlog, so it needs to be linked (or we can bundle our own fmtlib by coping fmt from system includedir inside spdlog at installation time, but is too ugly I think)

Reply:
This is really weird. Doesn't it mean that all Gentoo revdeps will have to explicitly link against libfmt?

Reply:
Correct, for this reason I didn't liked the idea of having fmtlib unbundled by default (I would prefer having a n optional system-fmtlib USE flag).

Reply:
Hmm, maybe libfmt should also install the .cc files and support header-only mode. After all, the headers have those `#ifdef`s and I suppose they'll fail when some package defines them.

Reply:
To what failures do you refer?

Reply:
Hmm, I must've been looking wrong. Ok, so what would happen if you `#define FMT_HEADER_ONLY` for the external libfmt case?

Reply:
C++ source files are installed by libfmt ebuild, then inside the header files of libfmt  it includes the source file like
```cpp
#ifdef FMT_HEADER_ONLY
# include "printf.cc"
#endif
```
so nothing will break. If FMT_HEADER_ONLY is defined then it's not needed to link against fmtlib shared library.

Reply:
Yes, so can we do that? Then spdlog could use external libfmt the same way as it uses internal, and therefore we avoid breaking reverse dependencies.

Reply:
Yes, and seems a good idea to have FMT_HEADER_ONLY by default. Also, when defining SPDLOG_FMT_EXTERNAL it should undef FMT_HEADER_ONLY.

Tomorrow I will do the needed changes.